From 1d754e1ba6d550dda302b79237bbfa691ca41df6 Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Sun, 3 Dec 2017 11:40:25 -0800
Subject: [PATCH 329/329] arm64: dts: allwinner: a64: enable sound on Pinebook

This commit enables I2S, digital and analog parts of audiocodec on
Pinebook

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 .../dts/allwinner/sun50i-a64-pinebook.dts     | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts
index 11518740ccca..d9bc63f9745d 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts
@@ -22,6 +22,11 @@
 		ethernet0 = &rtl8723cs;
 	};
 
+	amplifier: speaker-amplifier {
+		compatible = "simple-audio-amplifier";
+		enable-gpios = <&pio 7 7 GPIO_ACTIVE_HIGH>;
+	};
+
 	backlight: backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm 0 50000 0>;
@@ -66,10 +71,26 @@
 	};
 };
 
+&codec {
+       status = "okay";
+};
+
+&codec_analog {
+	status = "okay";
+};
+
 &cpu0 {
 	cpu-supply = <&reg_dcdc2>;
 };
 
+&dai {
+       status = "okay";
+};
+
+&ehci0 {
+	status = "okay";
+};
+
 &de {
 	status = "okay";
 };
@@ -323,6 +344,11 @@
 	};
 };
 
+&sound {
+	simple-audio-card,aux-devs = <&codec_analog>, <&amplifier>;
+	status = "okay";
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_pins_a>;
-- 
2.18.0

